taxi/tests/test_views.py

```diff
@@ -0,0 +1,163 @@
+
+
+# Index page tests
+class PublicIndexTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+
+    def test_login_required(self):
+        resp = self.client.get(INDEX_URL)
+
+        self.assertNotEqual(resp.status_code, 200)
+
+
+class PrivateIndexTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password123"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_index(self):
+        resp = self.client.get(INDEX_URL)
+
+        self.assertEqual(resp.status_code, 200)
+
+
+# Manufacturers page tests
+class PublicManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+
+    def test_login_required(self):
+        resp = self.client.get(MANUFACTURER_URL)
+
+        self.assertNotEqual(resp.status_code, 200)
+
+
+class PrivateManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password123"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="Test1", country="country1")
+        Manufacturer.objects.create(name="Test2", country="country2")
+
+        resp = self.client.get(MANUFACTURER_URL)
+        manufacturers = Manufacturer.objects.all()
+
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(
+            list(resp.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(resp, "taxi/manufacturer_list.html")
+
+
+# Car page tests
+class PublicCarTests(TestCase):
+    def setUp(self) -> None:
+        self.client = Client()
+
+    def test_login_required(self):
+        resp = self.client.get(CAR_URL)
+
+        self.assertNotEqual(resp.status_code, 200)
+
+
+class PrivateCarTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password123"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_cars(self):
+        manufacturer1 = Manufacturer.objects.create(
+            name="Test1",
+            country="country1"
+        )
+        manufacturer2 = Manufacturer.objects.create(
+            name="Test2",
+            country="country2"
+        )
+        Car.objects.create(
+            model="test_model1",
+            manufacturer=manufacturer1
+        )
+        Car.objects.create(
+            model="test_model2",
+            manufacturer=manufacturer2
+        )
+
+        resp = self.client.get(CAR_URL)
+        cars = Car.objects.all()
+
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(
+            list(resp.context["car_list"]),
+            list(cars)
+        )
+        self.assertTemplateUsed(resp, "taxi/car_list.html")
+
+
+# Driver page tests

```

Comment:
this comment is extra because the name of the test should be descriptive itself.
If you want to provide more information you can use docstrings

Reply:
fix everywhere